tests/post/steps/test_authentication.py

```diff
@@ -81,26 +81,45 @@ def control_plane_ip(host):
 
 # Given {{{
 
+
 @given(parsers.parse(
     "the control-plane Ingress path '{path}' is available"))
-def check_cp_ingress_pod(request, host, k8s_client, control_plane_ip):
+def check_cp_ingress_pod_and_container(
+    request,
+    host,
+    k8s_client,
+    control_plane_ip
+):
     ssh_config = request.config.getoption('--ssh-config')
     label = "release=nginx-ingress-control-plane"
+    namespace = "metalk8s-ingress"
+
+    def _wait_for_ingress_pod_and_container():
+        try:
+            pods = kube_utils.get_pods(
+                k8s_client, ssh_config, label, namespace=namespace
+            )
+        except Exception as exc:
+            pytest.fail("unable to get pods with error: {}".format(exc))
+
+        assert len(pods) > 0, (
+            "Expected only one Pod with label {}, but found {}"
```

Comment:
For me, the assertion and the message are not consistent, either we have:
```
assert pods, ("No Pod with label {} found").format(label)
```
Or:
```
assert len(pods) == 1, (
    "Expected only one Pod with label {}, but found {}"
).format(label, len(pods))
```

Reply:
Unfortunately, you should have picked the first option. We can't know how many Pods we'll find, as it depends on the number of `master` Nodes in the cluster.